src/rlookup.c

```diff
@@ -890,13 +903,22 @@ static void RLookup_HGETALL_scan_callback(RedisModuleKey *key, RedisModuleString
   RLookup_HGETALL_privdata *pd = privdata;
   size_t fieldCStrLen;
   const char *fieldCStr = RedisModule_StringPtrLen(field, &fieldCStrLen);
-  RLookupKey *rlk = RLookup_GetKeyEx_TEMP(pd->it, fieldCStr, fieldCStrLen, RLOOKUP_F_OCREAT | RLOOKUP_F_NAMEALLOC);
-  if (!pd->options->noSortables && (rlk->flags & RLOOKUP_F_SVSRC)) {
-    return;  // Can load it from the sort vector on demand.
+  RLookupKey *rlk = RLookup_FindKey(pd->it, fieldCStr, fieldCStrLen);
+  if (!rlk) {
+    // First returned document, create the key.
+    uint32_t flags = pd->options->noSortables ? RLOOKUP_F_NAMEALLOC | RLOOKUP_F_FORCE_LOAD : RLOOKUP_F_NAMEALLOC;
+    rlk = RLookup_GetKey_LoadEx(pd->it, fieldCStr, fieldCStrLen, fieldCStr, flags);
+    if (!rlk) {
+      return; // Key is sortable, can load it from the sort vector on demand.
+    }
+  } else if ((rlk->flags & RLOOKUP_F_QUERYSRC) ||
+             (!pd->options->noSortables && rlk->flags & RLOOKUP_F_SVSRC && !(rlk->flags & RLOOKUP_F_VAL_AVAILABLE))) {
+    return; // Key name is already taken by a query key, or it's already loaded.
```

Comment:
"or it's already loaded" means it is in Sorting Vector and not available (not UNF)? or in Sorting Vector and available (UNF)?

Reply:
Already loaded by some other loader, like if we first sort by a not sortable field and then request to load all the other fields as well